core/arch/arm/kernel/entry_a64.S

```diff
@@ -143,6 +148,13 @@
 		isb
 	.endm
 
+	.macro init_pan
+		mrs	x0, sctlr_el1
+		bic	x0, x0, #SCTLR_SPAN
+		msr	sctlr_el1, x0
+		.inst	0xd500409f | (0x1 << 8) /* Set PSTATE.PAN */
```

Comment:
Please let the assembler translate the instruction into machine code instead.

Reply:
Unfortunately the assembler is not able to translate `msr pan, x0`, etc. for now. The compiler would output error like `core/arch/arm/kernel/entry_a64.S:194: Error: selected processor does not support PSTATE field name 'pan'`

Reply:
Please add
```
        .macro write_pan reg
        /* msr pan, \reg */
        msr     S3_0_c4_c2_3, \reg
        .endm

        .macro write_pan_enable
        /* msr pan, #1 */
        msr     S0_0_c4_c1_4, xzr
        .endm

        .macro write_pan_disable
        /* msr pan, #0 */
        msr     S0_0_c4_c0_4, xzr
        .endm
```
to `core/arch/arm/include/arm64_macros.S` and use those macros instead.